src/SFML/Graphics/Shader.cpp

```diff
@@ -40,6 +40,18 @@
 
 #ifndef SFML_OPENGL_ES
 
+#if defined(SFML_SYSTEM_MACOS) || defined(SFML_SYSTEM_IOS)
+
+    #define castToGlHandle(x) reinterpret_cast<GLEXT_GLhandle>(static_cast<ptrdiff_t>(x))
+    #define castFromGlHandle(x) static_cast<unsigned int>(reinterpret_cast<ptrdiff_t>(x))
+
+#else
+
+    #define castToGlHandle(x) (x)
+    #define castFromGlHandle(x) (x)
+
+#endif
```

Comment:
:grin:

Like I said, `castToGlHandle` has to have the signature
```cpp
GLEXT_GLhandle castToGlHandle(unsigned int)
```
in both cases. Your gist still doesn't do this. Those macros aren't for casting from `void*` to anything else or back. They are only there to cast from `unsigned int` to `GLEXT_GLhandle` and vice versa. It's the `GLEXT_GLhandle` type that changes and not the function signature.

In essence, you would have to have 2 functions both with the same name and taking the same parameter type but returning a different type depending on system. Like I said, I don't think this is doable with templates alone.